### PR TITLE
fix(ingest/mssql): remove lower() method from sql_common get_db_name

### LINKED ISSUE #9857: convert_urns_to_lowercase: inconsistency of usage in SQLAlchemySource.get_db & get_identifier methods
**Describe the bug**
Link to slack [thread](https://datahubspace.slack.com/archives/C029A3M079U/p1707739499808849).

**To Reproduce**
Using 
https://github.com/datahub-project/datahub/blob/709c596867a17f8324a366b5bbec6a9002a4af7c/metadata-ingestion/src/datahub/ingestion/source/sql/sql_common.py#L392-L400
can lead to unexpected behaviour of urn creating. Despite of convert_urns_to_lowercase = True or False - database name will be in lowercase anyway.

**Additional context**
Investigating of relations between configs & modules leaded to next additional questions:
1.
https://github.com/datahub-project/datahub/blob/709c596867a17f8324a366b5bbec6a9002a4af7c/metadata-ingestion/src/datahub/ingestion/source/sql/mssql/source.py#L64 

has attribute 

https://github.com/datahub-project/datahub/blob/709c596867a17f8324a366b5bbec6a9002a4af7c/metadata-ingestion/src/datahub/ingestion/source/sql/mssql/source.py#L98-L101

but at the same moment LowerCaseDatasetUrnConfigMixin also have this attribute 

https://github.com/datahub-project/datahub/blob/709c596867a17f8324a366b5bbec6a9002a4af7c/metadata-ingestion/src/datahub/configuration/source_common.py#L57-L61

So, is this mistake or those attributes should perform different actions? If yes, why they have same name? If no, can we get rid off it from child class?

2.
When I`m trying to find explicit using of this attributes in code, I see only next things:

a)
https://github.com/datahub-project/datahub/blob/045c76a0e4014ce159acdb0ddfb0f2e1220c7c40/metadata-ingestion/src/datahub/ingestion/source/sql/mssql/source.py#L663

b)
https://github.com/datahub-project/datahub/blob/045c76a0e4014ce159acdb0ddfb0f2e1220c7c40/metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_utils.py#L177

c)
https://github.com/datahub-project/datahub/blob/045c76a0e4014ce159acdb0ddfb0f2e1220c7c40/metadata-ingestion/src/datahub/ingestion/api/source.py#L208-L229

I did not check it yet, but does it mean, that we can rid off all explicit using of convert_urns_to_lowercase attribute, because this parent class have method:

https://github.com/datahub-project/datahub/blob/045c76a0e4014ce159acdb0ddfb0f2e1220c7c40/metadata-ingestion/src/datahub/ingestion/api/source_helpers.py#L183-L197 

Or, if take a look at this from opposite point of view, I see only two explicit usage of convert_urns_to_lowercase attribute: in SQLServerSource and SnowflakeCommonMixin. Does it mean, that in all others sources this mechanizm (converting urns to lowercase) does not work?

@hsheth2 Can you, please, look at it one more time and after it I will create PR according to Slack thread. Thanks.

### PR DESCRIPTION
**Pull request by sleeperdeep**
Closes #9857

### CODE REVIEW

**Comment by sleeperdeep** [comment]
@mayurinehate @hsheth2 
Sorry guys, I made mistake and discarded all commits from previous PR. I created new one.

File: docs/how/updating-datahub.md [hunk 1]
```diff
@@ -19,6 +19,17 @@ This file documents any backwards-incompatible changes in DataHub and assists pe
 ## Next
 
 ### Breaking Changes
+- #9857 (#10773) `lower` method was removed from `get_db_name` of `SQLAlchemySource` class. This change will affect the urns of all related to `SQLAlchemySource` entities.
+  
+  Old `urn`, where `data_base_name` is `Some_Database`:
+  ```
+  - urn:li:dataset:(urn:li:dataPlatform:paltform,some_database.schema.entitie,ENV)
```
Comment (mayurinehate) [inline docs/how/updating-datahub.md:26]:
```suggestion
  - urn:li:dataJob:(urn:li:dataFlow:(mssql,demodata.Foo.stored_procedures,PROD),Proc.With.SpecialChar)
```

File: docs/how/updating-datahub.md [hunk 2]
```diff
@@ -19,6 +19,17 @@ This file documents any backwards-incompatible changes in DataHub and assists pe
...
+  ```
+  New `urn`, where `data_base_name` is `Some_Database`:
+  ```
+  - urn:li:dataset:(urn:li:dataPlatform:paltform,Some_Database.schema.entitie,ENV)
```
Comment (mayurinehate) [inline docs/how/updating-datahub.md:30]:
```suggestion
  - urn:li:dataJob:(urn:li:dataFlow:(mssql,DemoData.Foo.stored_procedures,PROD),Proc.With.SpecialChar)
```

File: docs/how/updating-datahub.md (hunk 1 above)
Comment (mayurinehate) [reply docs/how/updating-datahub.md:26]: Let's change the example to be of dataJob as only dataJob and container URNs have changed and not that of dataset.

**Review by mayurinehate — COMMENTED** [review]
Mostly looks good. Minor change suggested in example in doc.

File: docs/how/updating-datahub.md (hunk 1 above)
Comment (sleeperdeep) [reply docs/how/updating-datahub.md]: Fixed.

File: docs/how/updating-datahub.md (hunk 2 above)
Comment (sleeperdeep) [reply docs/how/updating-datahub.md]: Fixed

**Review by mayurinehate — APPROVED** [review]
LGTM

**Comment by sleeperdeep** [comment]
@mayurinehate Thanks.
@hsheth2 sorry for my mistake, can you, please, look at this PR one more time?

**Review by hsheth2 — APPROVED** [review]

**Review by hsheth2 — APPROVED** [review]

**Comment by sleeperdeep** [comment]
@hsheth2 @mayurinehate 
Hi guys! Can we merge this changes or something is not ok with it?

**Comment by sleeperdeep** [comment]
Reopen back

**Comment by sleeperdeep** [comment]
@hsheth2 @mayurinehate 
is failure of 'documentation / gh-pages'  known issue or I should fix some else with current PR?

**Comment by hsheth2** [comment]
@sleeperdeep the gh-pages one looked unrelated, so I've merged this in.

**Comment by maggiehays** [comment]
@coderabbitai summary

**Pull request merged**